src/main/java/com/epam/training/auction_backend/services/AuctionsServiceImpl.java

```diff
@@ -43,14 +47,14 @@ public void addAuction(AuctionTransferObject auctionTransferObject) {
     public Optional<AuctionTransferObject> getAuctionById(long id) {
         Auction auction = auctionRepository.findOne(id);
         Optional<AuctionTransferObject> auctionTransferObject;
+
         if (auction == null) {
-            auctionTransferObject = Optional.empty();
+            LOGGER.error(format(AUCTION_NOT_FOUND_MESSAGE, id));
```

Comment:
You should either log the exception or throw it. Now, it would be double-printed.